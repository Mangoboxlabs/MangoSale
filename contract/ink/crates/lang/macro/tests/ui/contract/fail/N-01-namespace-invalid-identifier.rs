use ink_lang as ink;

#[ink::contract]
mod invalid_namespace_identifier {
    #[ink(storage)]
    pub struct MyStorage {}

    #[ink(namespace = "::invalid_identifier")]
    impl MyStorage {
        #[ink(constructor)]
        pub fn constructor() -> Self {
            Self {}
        }

        #[ink(message)]
        pub fn message(&self) {}
    }
}

fn main() {}
