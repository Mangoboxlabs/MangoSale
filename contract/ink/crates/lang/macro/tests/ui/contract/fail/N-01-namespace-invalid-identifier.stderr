error: encountered invalid Rust identifier for namespace argument
 --> $DIR/N-01-namespace-invalid-identifier.rs:8:23
  |
8 |     #[ink(namespace = "::invalid_identifier")]
  |                       ^^^^^^^^^^^^^^^^^^^^^^
